<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>1. Identify dynamic differentially expressed genes (DDEGs) and generate master.list object</title>

<script src="data:application/javascript;base64,Ly8gUGFuZG9jIDIuOSBhZGRzIGF0dHJpYnV0ZXMgb24gYm90aCBoZWFkZXIgYW5kIGRpdi4gV2UgcmVtb3ZlIHRoZSBmb3JtZXIgKHRvCi8vIGJlIGNvbXBhdGlibGUgd2l0aCB0aGUgYmVoYXZpb3Igb2YgUGFuZG9jIDwgMi44KS4KZG9jdW1lbnQuYWRkRXZlbnRMaXN0ZW5lcignRE9NQ29udGVudExvYWRlZCcsIGZ1bmN0aW9uKGUpIHsKICB2YXIgaHMgPSBkb2N1bWVudC5xdWVyeVNlbGVjdG9yQWxsKCJkaXYuc2VjdGlvbltjbGFzcyo9J2xldmVsJ10gPiA6Zmlyc3QtY2hpbGQiKTsKICB2YXIgaSwgaCwgYTsKICBmb3IgKGkgPSAwOyBpIDwgaHMubGVuZ3RoOyBpKyspIHsKICAgIGggPSBoc1tpXTsKICAgIGlmICghL15oWzEtNl0kL2kudGVzdChoLnRhZ05hbWUpKSBjb250aW51ZTsgIC8vIGl0IHNob3VsZCBiZSBhIGhlYWRlciBoMS1oNgogICAgYSA9IGguYXR0cmlidXRlczsKICAgIHdoaWxlIChhLmxlbmd0aCA+IDApIGgucmVtb3ZlQXR0cmlidXRlKGFbMF0ubmFtZSk7CiAgfQp9KTsK"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="data:text/css,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">1. Identify dynamic differentially expressed genes (DDEGs) and generate master.list object</h1>



<div id="load-trendcatcher-r-package." class="section level4">
<h4>1. Load TrendCatcher R package.</h4>
</div>
<div id="read-in-the-demo-count-table." class="section level4">
<h4>2. Read in the demo count table.</h4>
<p>TrendCatcher <strong>requires</strong> the input count table in CSV file and with the column name in format as “ProjectName_Time_Rep1”. And the first column as gene symbol or gene ensembl ID.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>example.file.path<span class="ot">&lt;-</span><span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;Brain_DemoCountTable.csv&quot;</span>, <span class="at">package =</span> <span class="st">&quot;TrendCatcher&quot;</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>tb<span class="ot">&lt;-</span><span class="fu">read.csv</span>(example.file.path, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(tb)</span></code></pre></div>
<pre><code>##                    B_0_Rep1 B_0_Rep2 B_0_Rep3 B_0_Rep4 B_0_Rep5 B_6_Rep1
## ENSMUSG00000000001      118      104      115      119       98      127
## ENSMUSG00000000028        2        3        3        2        3        3
## ENSMUSG00000000031        2        1        2        2        2        2
## ENSMUSG00000000037        1        1        1        0        2        1
## ENSMUSG00000000056       57       48       41       51       44       45
## ENSMUSG00000000058      215      315      226      314      220       49
##                    B_6_Rep2 B_6_Rep3 B_6_Rep4 B_6_Rep5 B_24_Rep1 B_24_Rep2
## ENSMUSG00000000001      163      140      167       68       178       182
## ENSMUSG00000000028        2        2        4        1         4         3
## ENSMUSG00000000031        1        2        5        1        10         5
## ENSMUSG00000000037        1        0        1        1         0         0
## ENSMUSG00000000056       37       44       52       56        30        38
## ENSMUSG00000000058       68       58       81      246        30       106
##                    B_24_Rep3 B_24_Rep4 B_48_Rep1 B_48_Rep2 B_48_Rep3 B_48_Rep4
## ENSMUSG00000000001       162       200       121       115       405       136
## ENSMUSG00000000028         4         3        11         7         5        15
## ENSMUSG00000000031         3         1         2         2         1         5
## ENSMUSG00000000037         0         1         1         1         1         1
## ENSMUSG00000000056        27        23        33        42        16        44
## ENSMUSG00000000058        58        75       204       187        92       139
##                    B_48_Rep5 B_48_Rep6 B_72_Rep1 B_72_Rep2 B_72_Rep3 B_72_Rep4
## ENSMUSG00000000001       193       201       127       151       118       125
## ENSMUSG00000000028         8         4        10        10        24        20
## ENSMUSG00000000031         1         2         2         2         2         9
## ENSMUSG00000000037         1         1         1         1         1         1
## ENSMUSG00000000056        28        32        48        46        47        47
## ENSMUSG00000000058        78       127       251       278       211       224
##                    B_168_Rep1 B_168_Rep2 B_168_Rep3 B_168_Rep4
## ENSMUSG00000000001         67        105         86         98
## ENSMUSG00000000028          2          2          2          3
## ENSMUSG00000000031          1          1          2          4
## ENSMUSG00000000037          1          1          1          1
## ENSMUSG00000000056         47         64         78         81
## ENSMUSG00000000058        253        159        311        354</code></pre>
</div>
<div id="run-trendcatcher-and-generate-master.list-object" class="section level4">
<h4>3. Run TrendCatcher and generate master.list object</h4>
<p>This function will take few minutes to finish running with multiple cores.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>example.file.path<span class="ot">&lt;-</span><span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;Brain_DemoCountTable.csv&quot;</span>, <span class="at">package =</span> <span class="st">&quot;TrendCatcher&quot;</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>master.list<span class="ot">&lt;-</span><span class="fu">run_TrendCatcher</span>(<span class="at">count.table.path =</span> example.file.path,</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="at">baseline.t =</span> <span class="dv">0</span>,</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="at">time.unit =</span> <span class="st">&quot;h&quot;</span>,</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="at">min.low.count =</span> <span class="dv">1</span>,</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="at">para.core.n =</span> <span class="cn">NA</span>,</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="at">dyn.p.thres =</span> <span class="fl">0.05</span>)</span></code></pre></div>
</div>
<div id="check-master.list-in-detail." class="section level4">
<h4>4. Check master.list in detail.</h4>
<p>To save time of running, we already put the output master.list object in the ‘/inst/extdata’ folder. You can simply load it into your environment.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>demo.master.list.path<span class="ot">&lt;-</span><span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;BrainMasterList.rda&quot;</span>, <span class="at">package =</span> <span class="st">&quot;TrendCatcher&quot;</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(demo.master.list.path)</span></code></pre></div>
<p><strong>First, check what elements are included in the master.list list object. </strong></p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(master.list)</span></code></pre></div>
<pre><code>## [1] &quot;time.unit&quot;    &quot;baseline.t&quot;   &quot;t.arr&quot;        &quot;Project.name&quot; &quot;raw.df&quot;      
## [6] &quot;fitted.count&quot; &quot;master.table&quot;</code></pre>
<ul>
<li>“time.unit” is the time unit, for example “h” represents hour.</li>
<li>“baseline.t” is the baseline time. Here it is hour 0.</li>
<li>“t.arr” is the time array in time course study. Here is 0, 6, 24, 48, 72, 168h.</li>
<li>“Project.name” is the project name. Here is “B”.</li>
<li>“raw.df” is the count table ordered by time and replicate ID.</li>
<li>“fitted.count” is the ANOVA smooth model fitted count number from replicates.</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">c</span>(master.list<span class="sc">$</span>time.unit, master.list<span class="sc">$</span>baseline.t))</span></code></pre></div>
<pre><code>## [1] &quot;h&quot; &quot;0&quot;</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>master.list<span class="sc">$</span>t.arr</span></code></pre></div>
<pre><code>## [1]   0   6  24  48  72 168</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>master.list<span class="sc">$</span>Project.name</span></code></pre></div>
<pre><code>## [1] &quot;B&quot;</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(master.list<span class="sc">$</span>raw.df)</span></code></pre></div>
<pre><code>##                    B_0_Rep1 B_0_Rep2 B_0_Rep3 B_0_Rep4 B_0_Rep5 B_6_Rep1
## ENSMUSG00000000001      118      104      115      119       98      127
## ENSMUSG00000000028        2        3        3        2        3        3
## ENSMUSG00000000031        2        1        2        2        2        2
## ENSMUSG00000000037        1        1        1        0        2        1
## ENSMUSG00000000056       57       48       41       51       44       45
## ENSMUSG00000000058      215      315      226      314      220       49
##                    B_6_Rep2 B_6_Rep3 B_6_Rep4 B_6_Rep5 B_24_Rep1 B_24_Rep2
## ENSMUSG00000000001      163      140      167       68       178       182
## ENSMUSG00000000028        2        2        4        1         4         3
## ENSMUSG00000000031        1        2        5        1        10         5
## ENSMUSG00000000037        1        0        1        1         0         0
## ENSMUSG00000000056       37       44       52       56        30        38
## ENSMUSG00000000058       68       58       81      246        30       106
##                    B_24_Rep3 B_24_Rep4 B_48_Rep1 B_48_Rep2 B_48_Rep3 B_48_Rep4
## ENSMUSG00000000001       162       200       121       115       405       136
## ENSMUSG00000000028         4         3        11         7         5        15
## ENSMUSG00000000031         3         1         2         2         1         5
## ENSMUSG00000000037         0         1         1         1         1         1
## ENSMUSG00000000056        27        23        33        42        16        44
## ENSMUSG00000000058        58        75       204       187        92       139
##                    B_48_Rep5 B_48_Rep6 B_72_Rep1 B_72_Rep2 B_72_Rep3 B_72_Rep4
## ENSMUSG00000000001       193       201       127       151       118       125
## ENSMUSG00000000028         8         4        10        10        24        20
## ENSMUSG00000000031         1         2         2         2         2         9
## ENSMUSG00000000037         1         1         1         1         1         1
## ENSMUSG00000000056        28        32        48        46        47        47
## ENSMUSG00000000058        78       127       251       278       211       224
##                    B_168_Rep1 B_168_Rep2 B_168_Rep3 B_168_Rep4
## ENSMUSG00000000001         67        105         86         98
## ENSMUSG00000000028          2          2          2          3
## ENSMUSG00000000031          1          1          2          4
## ENSMUSG00000000037          1          1          1          1
## ENSMUSG00000000056         47         64         78         81
## ENSMUSG00000000058        253        159        311        354</code></pre>
<p><strong>The fitted.count table each column represents.</strong></p>
<ul>
<li>Gene, gene ensembl ID or gene symbol from the raw data.</li>
<li>Time, time measured.</li>
<li>Fit.Count, the ANOVA smooth model fitted count value.</li>
<li>mu, the estimated mean count of baseline expression.</li>
<li>disp, dispersion estimated of the baseline expression.</li>
<li>t.p.val, the dynamic p-value at each time point compared to baseline NB distribution.</li>
<li>dyn.p.val, the combined dynamic p-value from all time points.</li>
<li>dyn.p.val.adj, the adjusted dynamic p-value from all DDEGs.</li>
</ul>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(master.list<span class="sc">$</span>fitted.count)</span></code></pre></div>
<pre><code>##                 Gene Time Fit.Count    mu     disp    t.p.val dyn.p.val
## 1 ENSMUSG00000000001    0 110.80002 110.8 12.88174 1.00000000  8.67e-03
## 2 ENSMUSG00000000001    6 137.81128 110.8 12.88174 0.19627656  8.67e-03
## 3 ENSMUSG00000000001   24 175.52409 110.8 12.88174 0.03646947  8.67e-03
## 4 ENSMUSG00000000001   48 185.67185 110.8 12.88174 0.02176459  8.67e-03
## 5 ENSMUSG00000000001   72 139.55592 110.8 12.88174 0.18189626  8.67e-03
## 6 ENSMUSG00000000001  168  88.65132 110.8 12.88174 0.26173223  8.67e-03
##   dyn.p.val.adj
## 1    0.06409249
## 2    0.06409249
## 3    0.06409249
## 4    0.06409249
## 5    0.06409249
## 6    0.06409249</code></pre>
<p><strong>For the master table, each column represents.</strong></p>
<ul>
<li>Gene, gene ensembl ID or gene symbol.</li>
<li>pattern, the gene trajectory master-pattern type.</li>
<li>start.idx, the index of where the corresponding trend start in t.arr.</li>
<li>end.idx, the index of where the corresponding trend end in the t.arr.</li>
<li>dynTime, the time where t.p.val &lt;=0.05, the break point.</li>
<li>dynSign, “-” means down, “+” means up.</li>
<li>start.t, the time of where the corresponding trend start.</li>
<li>end.t, the time of where the corresponding trend end.</li>
<li>pattern_str, the string ready for print indicates the gene trajectory sub-pattern type.</li>
</ul>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(master.list<span class="sc">$</span>master.table)</span></code></pre></div>
<pre><code>##                 Gene  pattern start.idx end.idx     dynTime  dynSign start.t
## 1 ENSMUSG00000025283 up_down_      1_4_    4_6_ 6_24_48_72_ +_+_+_+_   0_48_
## 2 ENSMUSG00000028967 up_down_      1_2_    2_6_    6_24_48_   +_+_+_    0_6_
## 3 ENSMUSG00000039236 up_down_      1_2_    2_6_       6_24_     +_+_    0_6_
## 4 ENSMUSG00000078920 up_down_      1_2_    2_6_    6_24_48_   +_+_+_    0_6_
## 5 ENSMUSG00000105987 up_down_      1_4_    4_6_ 6_24_48_72_ +_+_+_+_   0_48_
## 6 ENSMUSG00000022221 up_down_      1_3_    3_6_ 6_24_48_72_ +_+_+_+_   0_24_
##     end.t         pattern_str dyn.p.val dyn.p.val.adj
## 1 48_168_ 0h_up_48h_down_168h  1.11e-16  4.615657e-13
## 2  6_168_  0h_up_6h_down_168h  1.11e-16  4.615657e-13
## 3  6_168_  0h_up_6h_down_168h  1.11e-16  4.615657e-13
## 4  6_168_  0h_up_6h_down_168h  1.11e-16  4.615657e-13
## 5 48_168_ 0h_up_48h_down_168h  2.22e-16  7.385052e-13
## 6 24_168_ 0h_up_24h_down_168h  3.33e-16  9.231315e-13</code></pre>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
